hydromt_sfincs/subgrid.py

```diff
@@ -403,7 +406,18 @@ def build(
                         f"WARNING: Interpolate data at {int(np.sum(np.isnan(da_dep.values)))} cells"
                     )
                     da_dep = da_dep.raster.interpolate_na(method="rio_idw")
-                assert np.all(~np.isnan(da_dep))
+
```

Comment:
werkt dit @keesnederhoff en @DirkEilander?

Ik dacht dat we dit hadden geprobeerd af te vangen met die rio_idw methode hierboven. Als dat nog niet afdoende was lijkt me dit prima (als het werkt), anders moeten we even een andere oplossing verzinnen voor extrapoleren.

Reply:
De onderstaande regels werken inderdaad niet, zie [hydromt docs](https://deltares.github.io/hydromt/latest/_generated/hydromt.DataArray.raster.interpolate_na.html#hydromt.DataArray.raster.interpolate_na)
```
da_dep = da_dep.raster.interpolate_na(
    fill_value="extrapolate", extrapolate=True
)
```

Wat wel zou kunnen werken (als de rio_idw methode inderdaad niet goed extrapoleert), is gebruik maken van de xarray `interpolate_na` methode, zie [xarray docs](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.interpolate_na.html). Die werkt echter maar in een dimensie tegelijkertijd, dus niet ideaal.
```
da_dep = da_dep.interpolate_na(
    dim=da_dep.raster.x_dim, fill_value="extrapolate", extrapolate=True
).interpolate_na(
    dim=da_dep.raster.y_dim, fill_value="extrapolate", extrapolate=True
)
```

We kunnen ook de hydromt methode uitbreiden met een extrapolate optie op basis van nearest als dat wenselijk is. Dan moeten we daar even een issue voor inschieten.

Reply:
Laten we voor nu even de xarray methode gebruiken dan, kunnen we later altijd nog vervangen